app/services/sso/openid_connect.rb

```diff
@@ -38,7 +38,14 @@ def bearer_token_set?
 
     def valid_issuer?
       payload = jwt_token.decoded_payload
-      payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
+      unless payload.key?('iss') && (payload['iss'] == Setting['oidc_issuer'])
+        Foreman::Logging.logger('app').error "Invalid OIDC issuer received in JWT."
```

Comment:
I'd still consider logging the issuer at debug level. This is based on my experience as an admin where it can be very frustrating to figure out exactly why something isn't working.
```ruby
logger = Foreman::Logging.logger('app')
logger.error "Invalid OIDC issuer received in JWT."
logger.debug "Received invalid OIDC issuer '#{payload['iss']}'"
```